config.mk

```diff
@@ -36,9 +37,9 @@ CONSOLE=stdio
 
 # WARNING: cannot use stdio or pty backend if any already used by console
 
-SERIAL=none
+#SERIAL=none
 #SERIAL=stdio
-#SERIAL=pty
+SERIAL=pty
```

Comment:
With CONSOLE=stdio and SERIAL=stdio, will implementation work using Linux/macOS "terminal" to communicate with emulated serial port when emulation running, and ">" etc commands when single stepping?

Reply:
Just try to build EMU86 with that config :yum:

Reply:
I see warning sign in above code that setting CONSOLE and SERIAL to stdio won't work. However, when emulating ELKS console through INT 10h, EMU86 redirects console output to stdio when CONSOLE=stdio. 

Wondering about similar capability when 8018X is effectively running ELKS "console" through serial port. That is, emulated serial port is treated as new option CONSOLE=serial.

Reply:
I'll try it, thanks.

Reply:
In the 8018X configuration, ELKS does not use any 'console device', but only the 'serial port device'. So configuring the console device back end to `stdio` would have no effect, even if EMU86 supports several `stdio` back end instances.

Reply:
Not true in current ELKS repo: the 8018X configuration implements a "headless" console, but 'console device' nonetheless, which happens to communicate through 8018X polling serial port, rather than through, say BIOS INT 16h/INT 10h. So ELKS thinks console is exactly the same and knows nothing about serial port.

There is some talk about actually implementing 8018X serial port as ELKS 'serial device', which has its own set of advantages/disadvantages: would allow treatment of 8018X serial port as /dev/ttyS0 device and then require CONFIG_CONSOLE_SERIAL in order to use it as console device. ELKS does not currently support "no" console.

So it is for these issues I bring up the question on emulation in EMU86.

Reply:
> Not true in current ELKS repo: the 8018X configuration implements a "headless" console, but 'console device' nonetheless, which happens to communicate through 8018X polling serial port

This is exactly what I said : there is no console **device** from emulation point of view : no keyboard, no display board, just a serial port device.

Reply:
Yes, I get it now, thank you. It would be nice to allow serial port to appear as console to EMU86 as future option, just as BIOS I/O can appear as console, to utilize the benefits in interacting with EMU86 using CONSOLE=stdio vs CONSOLE=pty. That can all wait for another day, if ever.

Reply:
I am afraid several terms are mixed here : the 'console' HW device (keyboard, display), the EMU86 'console' (that is always connected to the `stdio` of the EMU86 process) and the ELKS 'console', that is an abstraction, internally mapped in ELKS to a device, either directly ('direct console'), thought an API ('bios console'), or here redirected to the serial port ('8018X headless console').

You can use the EMU86 process `stdio` for emulated serial port I/O using `SERIAL=stdio`, but I noticed a side effect in that configuration, so the choice of `SERIAL=pty` to separate the two streams until that effect is fixed.